postgres.go

```diff
@@ -115,7 +115,30 @@ func (dialector Dialector) DefaultValueOf(field *schema.Field) clause.Expression
 
 func (dialector Dialector) BindVarTo(writer clause.Writer, stmt *gorm.Statement, v interface{}) {
 	writer.WriteByte('$')
```

Comment:
Since the prepare mode does not allow changes to the table structure, we specify the `QueryExecModeSimpleProtocol` mode in the migrator.
The problem here is that when there are parameters, we need to recalculate the position of the parameters.

refer to https://github.com/go-gorm/postgres/pull/109

Reply:
https://github.com/go-gorm/postgres/blob/e6d2435140dc0305dae1e61780f3a55cdced47c7/migrator.go#L588
I want let limit use binding parameter ，so the code(d.Statement.Vars = append([]interface{}{pgx.QueryExecModeSimpleProtocol}, d.Statement.Vars...))  will  leading the sql becoming  select * from objects limit $2 whitch can't be excuted right ,   expected sql is  "select * from objects limit $1".    BindVarTo method just need  parameters (d.Statement.Vars)  prepare the sql   except  parameters like pgx.QueryExecModeSimpleProtocol  . so  store the   parameters like pgx.QueryExecModeSimpleProtocol  in tempVars  , bind the sql var ,then restore the Statement.Vars
. so  these code can pass the tests

Reply:
I mean
1. We only need to modify `len(stmt.Vars)` in `writer.WriteString(strconv.Itoa(len(stmt.Vars)))`, no additional memory allocation is required.
2. We only need to handle the case where `stmt.Vars[0]` is equal to `pgx.QueryExecModeSimpleProtocol`, and do not need to loop through other cases that we do not use.

Reply:
fixed the problem,I really appriciate your time spant on reviewing code.  thanks!